packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -33,6 +36,17 @@ export function createAuthMiddleware({
       ) {
         ctx.throw(403, 'Insufficient Grant')
       }
+      const grantRef = await grantReferenceService.get(grant.grant)
+      if (grantRef) {
+        if (grantRef.clientId !== grant.clientId) {
+          ctx.throw(409, 'Unknown Client ID')
+        }
+      } else {
+        await grantReferenceService.create({
```

Comment:
I think the `get` and `create` should happen within a db transaction

Reply:
You are saying within the same transaction?

Reply:
Yes, otherwise `create` will throw if the grant was created (after `get`) via another another request